File: AnalyticHierarchyProcess/app/src/main/java/tr/edu/sakarya/analytichierarchyprocess/InputFilterMinMax.java
package tr.edu.sakarya.analytichierarchyprocess;

import android.text.InputFilter;
import android.text.Spanned;

public class InputFilterMinMax implements InputFilter {
    private int min;
    private int max;

    public InputFilterMinMax(int min, int max) {
        this.min = min;
        this.max = max;
    }

    @Override
    public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
        int input = Integer.parseInt(dest.toString() + source.toString());
        if (input >= min && input <= max)
            return null;

        return "";
    }
}

File: AnalyticHierarchyProcess/app/src/main/java/tr/edu/sakarya/analytichierarchyprocess/InputFilterMinMax.kt
package tr.edu.sakarya.analytichierarchyprocess

import android.text.InputFilter
import android.text.Spanned

class InputFilterMinMax(private val min: Int, private val max: Int) : InputFilter {

    override fun filter(
        source: CharSequence,
        start: Int,
        end: Int,
        dest: Spanned,
        dstart: Int,
        dend: Int
    ): CharSequence? {
        val input = (dest.toString() + source.toString()).toIntOrNull()
        return if (input in min..max) null else ""
    }
}
